march_hardware/src/MarchRobot.cpp

```diff
@@ -34,28 +34,40 @@ MarchRobot::MarchRobot(::std::vector<Joint> jointList, urdf::Model urdf, PowerDi
 
 void MarchRobot::startEtherCAT(bool reset_imc)
 {
-  if (!hasValidSlaves())
+  bool sw_reset = false;
+  for (int i = 0; i < 5; i++)
```

Comment:
Where does the five come from?

Reply:
It was arbitrary. We should at least try a few times to recover the system, once might just not be enough.

Reply:
It feels a bit like restarting iMC' s until they work. Anyone else any thoughts?

Reply:
I think this is not necessary since we have not seen this fail even once last Wednesday. Hence, I think it is better to remove it for now, we can always re-implement it if it turns out to be necessary.

Reply:
Is there a situation where it would not work the first time but it does the second time? I guess if etherCAT does not start successfully the first time we should not try this again but display the error which has occurred?

Reply:
I guess I'll remove this for now